### ISSUE TITLE
Allows mods to request a separate list of save files

### ISSUE THREAD

**Issue by EvelynTSMG** [peppy-enterprises/fahrenheit#30]
For mods like Kari's Masters Challenge, which alter the game in significant ways and would prefer to not force the player to remember which save file is which.
Save files should be stored in folders, possibly inside of the standard save file path.
If multiple mods request separate save file list at the same time, either one should be prioritized through load order, or the combination of those mods should have its own save file list.

**Comment by fkelava** [comment]
Notes:
- This should be a manifest-level flag.
- The list is scoped by _the union_ of all mods that declare this manifest-level flag.
- It is unclear where we have to hook to intercept the save list presentation. ``SaveDataManager`` and methods from its vtable are a good start.
- This should also (in one fell swoop) eliminate the arbitrary 200 save file limit. It is yet to be understood whether this limit is actually _arbitrary_.

**Comment by fkelava** [comment]
It has been noted that we should pair this with callbacks that allow mods to retrieve or save some internal state at save/load time.

**Comment by fkelava** [comment]
This is blocked by an unsuccessful hook to the save enumeration function at +2F0DA0 and/or a lack of understanding of the data structure the game expects this function to produce.

**Comment by EvelynTSMG** [comment]
Separate save lists will be scoped by a hash, created from the list of mods requiring separate saves, sorted by load order. Additionally, we have decided to include another layer of separation — _save sets_. The user will be able to create or delete them through the mod manager, and switch between them in-game. Fahrenheit will remember which set was last active between boots.

The game's save file limit of 200 will be increased. Currently, it has been increased to 500, but this may be changed before release.

Unfortunately, due to the game's implementation of the save/load menu, switching between sets causes tremendous slowdowns over time. This has forced us to reimplement the save/load menu ourselves.

This reimplementation's UI/UX will be low priority at first, but I am willing to work on it myself once the necessary issues are resolved (that is, once we are able to render the game's textures on our own — #55.)
